spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java

```diff
@@ -188,6 +188,9 @@ public RabbitTemplate rabbitTemplate(ConnectionFactory connectionFactory) {
 			if (templateProperties.getReplyTimeout() != null) {
 				rabbitTemplate.setReplyTimeout(templateProperties.getReplyTimeout());
 			}
+			rabbitTemplate.setExchange(templateProperties.getExchange());
+			rabbitTemplate.setRoutingKey(templateProperties.getRoutingKey());
+			rabbitTemplate.setChannelTransacted(templateProperties.isChannelTransacted());
```

Comment:
I would prefer the use of null checks rather than "mirroring" the current defaults in the template - if we ever change the defaults in the template they will be undone here. See the properties above (reply timeout etc) for examples; so `channelTransacted` should be `Boolean`  so we can check whether the user has explicitly set the property.

While it is unlikely those defaults will change, this is the general pattern used by Boot's auto configuration.

Reply:
> this is the general pattern used by Boot's auto configuration.

Actually, that's not 100% accurate. When we can add a test  that validates that our default are the same as the underlying tech's default, we try as much as we can to put it back in the config. That way, users can know what the defaults are in the IDE.

Reply:
OK; that makes sense.

As discussed offline; if we expose the `channelTransacted` flag on the template, we should do the same for the listener container factories so that any rabbit template operations performed on the container thread will participate in the same transaction.

Reply:
I will leave these then since they are tested.